v2/message.go

```diff
@@ -441,6 +442,11 @@ func SetExpectedOffset(expectedOffset int64) MessageOption {
 // NewMessage returns a serialized message for the given payload and options.
 func NewMessage(value []byte, options ...MessageOption) []byte {
 	opts := &MessageOptions{Headers: make(map[string][]byte)}
+	// set expected offset to -1 to inicate next offset.
+	// this will keep clients that are not yet providing support for Optimistic Concurrency Control
+	// to operate normally.
+	opts.ExpectedOffset = -1
```

Comment:
This needs to get set in `Publish` and `PublishAsync` as well.

Reply:
updated